### ISSUE TITLE
ability to assign users permissions to add/edit/remove documents

### ISSUE THREAD

**Issue by SergioCBPC** [roman-yagodin/R7.Documents#62]
Not sure if this should be a feature request or a bug report.
I was under the impression that I would be able to assign permissions to a role or user so that the user would be able to add/edit/remove documents from the module without giving the user edit permissions on the module or page. So far it seems that only admins can add/edit content in the module.
I thought it was maybe the theme that wasn't displaying the controls but I tried this with a bootstrap3 theme and the default dnn themes on a 9.1.0 install.
Please advise if I'm doing something wrong to accomplish this but my guess is that this is by design. Thus, I really hope that in the near future this feature gets added. as it becomes somewhat limited to having to do tasks that users with content editing permissions could be doing instead of the admin.

**Comment by roman-yagodin** [comment]
Thanks for a feedback, @SergioCBPC!

Yes, it is by design, at least for now. Setting permissions for individual documents adds serious overhead and I'm not sure is it worth to implement or not.

I do have a plan to implement sharing documents between the modules - among other things, it will allow to create some private pages for content editors with the modules containing only documents the content editors can manage, and some kind of aggregator module on a public page containing all documents from those modules.

If you want to have more control over module/page permissions, you can look at this: https://github.com/ohine/Dnn.Enhanced-Permissions-Provider

**Comment by improwise** [comment]
I believe that whta SergioCBPC is askin for is not permissions on document levels but rather that it would be nice if a user/role could add/delete/change documents without being able to change the module itself.

**Comment by SergioCBPC** [comment]
That is correct improwise. While having permissions at a document level would be awesome, what I was really asking for is the ability for any user/role be able to add/delete/change documents without having to give the user permissions to edit the module. For example, in the Form and List module you can give permission to a specific user/role so that this user/role can create/edit/modify records on a list without giving the user/role edit permissions on the module. This way, Some users can have just view permisisons and others can add/edit/change records but cannot go into the module settings and make changes to the module itself.

**Comment by roman-yagodin** [comment]
Sorry for the late response, but it looks like it would be sufficient for the described use case to just disallow editing of critical module settings for regular editors - please correct me if I'm wrong.
